components/table/Sidebar.jsx

```diff
@@ -0,0 +1,48 @@
+import React, { useRef } from 'react'
+
+import tableClassNames from './index.css'
+
+import { CloseIcon } from 'design'
+
+class Sidebar extends React.Component {
+  constructor(props) {
+    super(props)
+    const { children } = props
+    if (children) React.Children.only(children)
+  }
+
+  render() {
+    const { isOpen, children, ...passProps } = this.props
+    return (
+      <div className={tableClassNames.sidebar}>
+        {children && React.cloneElement(children, passProps)}
+      </div>
+    )
+  }
+}
+
+Sidebar.displayName = 'TableSidebar'
+
+const SidebarHeader = ({ children, ...passProps }) => {
+  const closeIconRef = useRef(null)
+
+  return (
+    <div {...passProps}>
+      {children}
+      <CloseIcon
+        ref={closeIconRef}
+        aria-label="Close table sidebar"
+        onClick={() =>
+          closeIconRef.current.dispatchEvent(
+            new Event('sidebar-close', { bubbles: true })
+          )
+        }
```

Comment:
Why do you create a custom event instead using standard click event?

Reply:
How would I distinguish this event from row click? I would need to pass some ID or className and it's almost the same as this one. :)

Reply:
It's not correct that you render details window right in the table. The point of redesign was that it's not neither logical nor semantically correct to have not-a-row in the table.

Reply:
Sorry I was not clear. Sidebar is not a part of <table>. Table component consist of two divs. One contains table element and one contains sidebar.